### DISCUSSION TITLE
Should pendingElement short circuit route.pending

### DISCUSSION THREAD

**Discussion by fetters5** [TanStack/router#186] [Q&A]
Hey I am not sure if this would be considered a bug or a feature thus the discussion.

I see that on routes that have a pending element  the global router.pending  flag will end as soon as the pending element shows.  So if I have a global pending indicator similar to https://react-location.tanstack.com/guides/pending-states#global-pending-navigation-indicators it will stop early and the pending element will show.  Shouldnt the global router.pending only end when the loader is resolved or rejected?

Here is a link to the sandbox https://codesandbox.io/s/condescending-flower-b1prw?file=/src/index.tsx and preview https://b1prw.csb.app/

**Comment by tannerlinsley** [comment]
I think you bring up a good point. `router.pending` is referring strictly to the suspended state of the route transition itself, not the actual sub matches that might be loading/pending/resolves/rejected.

I think a good path forward would be this: https://codesandbox.io/s/withered-haze-ypwte?file=/src/index.tsx:2510-2632

**Reply by tannerlinsley** [reply]
If you think that's a good solution, I would love it if you opened a PR to add it to the docs where you think it would fit best?
